app/controllers/v0/facilities/ccp_controller.rb

```diff
@@ -14,13 +30,60 @@ def show
   end
 
   def services
-    ppms = Facilities::PPMS::Client.new
     result = ppms.specialties
     render json: result
   end
 
   private
 
+  def provider_locator
+    providers = ppms.provider_locator(params)
+    total = providers.length
+    start_ind = (page - 1) * BaseFacility.per_page
+    providers = providers[start_ind, BaseFacility.per_page - 1]
+    providers.map! do |provider|
+      prov_info = ppms.provider_info(provider['ProviderIdentifier'])
+      provider.add_details(prov_info)
+      provider
+    end
+    pages = { current_page: page, per_page: BaseFacility.per_page,
+              total_pages: total / BaseFacility.per_page + 1, total_entries: total }
+
+    render json: providers,
+           each_serializer: ProviderSerializer,
+           meta: { pagination: pages }
+  end
+
+  def pos_locator(pos_code)
+    place_of_service = ppms.pharmacy_locator(params, pos_code)
+
+    pages = { current_page: page, per_page: BaseFacility.per_page,
+              total_pages: place_of_service.length / BaseFacility.per_page + 1, total_entries: total }
+
+    render json: providers,
+           each_serializer: ProviderSerializer,
+           meta: { pagination: pages }
+  end
+
+  def ppms
```

Comment:
What do you think about calling it `client` or `api`? Since its private I shouldn't care about what it is, and with it you get controller methods that look something like:

```
    result = api.specialties
    render json: result
```

or

```
    result = client.specialties
    render json: result
```

Just a thought

Reply:
I don't have any issues with either.